From 0a89668a07d873d1942f702a595f438025d526fd Mon Sep 17 00:00:00 2001
From: Mikolaj Izdebski <mizdebsk@redhat.com>
Date: Mon, 30 Sep 2013 11:05:41 +0200
Subject: [PATCH] Don't assume package has a single name

One package can have different names, for example "" and "__default".
---
 .../maven/installer/impl/DefaultInstaller.java            | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/xmvn-core/src/main/java/org/fedoraproject/maven/installer/impl/DefaultInstaller.java b/xmvn-core/src/main/java/org/fedoraproject/maven/installer/impl/DefaultInstaller.java
index a3e9f9b..9472c08 100644
--- a/xmvn-core/src/main/java/org/fedoraproject/maven/installer/impl/DefaultInstaller.java
+++ b/xmvn-core/src/main/java/org/fedoraproject/maven/installer/impl/DefaultInstaller.java
@@ -97,7 +97,7 @@ public class DefaultInstaller
 
     private Configuration configuration;
 
-    private Map<String, Package> packages;
+    private Map<Package, Package> packages;
 
     private PackagingRule ruleForArtifact( Artifact artifact )
     {
@@ -134,13 +134,12 @@ public class DefaultInstaller
         String packageName = rule.getTargetPackage();
         if ( StringUtils.isEmpty( packageName ) )
             packageName = Package.MAIN;
-        Package pkg = packages.get( packageName );
 
-        if ( pkg == null )
-        {
-            pkg = new Package( packageName, settings, logger );
-            packages.put( packageName, pkg );
-        }
+        Package pkg = new Package( packageName, settings, logger );
+        if ( packages.containsKey( pkg ) )
+            pkg = packages.get( pkg );
+        else
+            packages.put( pkg, pkg );
 
         if ( logger.isDebugEnabled() )
         {
@@ -464,7 +463,7 @@ public class DefaultInstaller
         packages = new TreeMap<>();
 
         Package mainPackage = new Package( Package.MAIN, settings, logger );
-        packages.put( Package.MAIN, mainPackage );
+        packages.put( mainPackage, mainPackage );
 
         Set<Artifact> artifactSet = request.getArtifacts();
 
-- 
1.8.3.1

